tests/Functional/src/com/microsoft/aad/adal/test/BrokerProxyTests.java

```diff
@@ -205,6 +208,175 @@ public void testCanSwitchToBrokerValid_Skip()
+        AuthenticationSettings.INSTANCE.setBrokerSignature(testTag);
+        Account[] accts = getAccountList("currentUserName", authenticatorType);
+        prepareProxyForTest(brokerProxy, authenticatorType, brokerPackage, contextPackage,
+                signature, true, accts);
+        ReflectionUtils.setFieldValue(brokerProxy, "mContext", mockContext);
+        
+        //set the access to private method
+        Method m = ReflectionUtils.getTestMethod(brokerProxy, "verifyManifestPermissions");
+        
+        //test@case permission GET_ACCOUNTS is missing
+        try {
+            m.invoke(brokerProxy);
+            Assert.fail("It is expected to return an exception here.");
+        } catch(InvocationTargetException e) {
+            assertTrue(e.getCause() instanceof UsageAuthenticationException);
+            assertTrue(((UsageAuthenticationException)e.getCause()).getMessage().toString().contains("[GET_ACCOUNTS]"));
+            assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING,
+                    ((UsageAuthenticationException)e.getCause()).getCode());
+        }
+    }
+    
+    public void testVerifyManifestPermission_missingPermission_MANAGE_ACCOUNTS_USE_CREDENTIALS() throws 
+            NoSuchAlgorithmException, NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, 
+            NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException, 
+            NoSuchFieldException, NameNotFoundException {
+        //mock Brokerproxy
+        FileMockContext mockContext = new FileMockContext(getContext());
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.GET_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        Object brokerProxy = ReflectionUtils.getInstance("com.microsoft.aad.adal.BrokerProxy");
+        String authenticatorType = AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE;
+        String brokerPackage = AuthenticationConstants.Broker.PACKAGE_NAME;
+        String contextPackage = "com.test";
+        Signature signature = new Signature(testSignature);
+        AuthenticationSettings.INSTANCE.setBrokerSignature(testTag);
+        Account[] accts = getAccountList("currentUserName", authenticatorType);
+        prepareProxyForTest(brokerProxy, authenticatorType, brokerPackage, contextPackage,
+                signature, true, accts);
+        ReflectionUtils.setFieldValue(brokerProxy, "mContext", mockContext);
+        
+        //set the access to private method
+        Method m = ReflectionUtils.getTestMethod(brokerProxy, "verifyManifestPermissions");
+        
+        //test@case permissions USE_CREDENTIALS and MANAGE_ACCOUNTS are missing
+        try {
+            m.invoke(brokerProxy);
+            Assert.fail();
+        } catch(InvocationTargetException e) {
+            assertTrue(e.getCause() instanceof UsageAuthenticationException);
```

Comment:
nit: 1. check the cause instance first
2. UsageAuthenticationException exception = (UsageAuthenticationException) e.getCause
3. reuse it in the later assert.